### PR TITLE
process: rewrite and simplify the issue_42 test

### PR DESCRIPTION
**Pull request by ipetkov**
## Motivation

The process_issue_42 test has been updated/refactored multiple times since its existence, and was a bit hard to follow (and potentially racy when running in the CI).

## Solution

This change rewrites the test to be more straightforward, and using the current APIs. The outcome is largely the same as before, except the test iterations are performed in parallel, and we do not enforce dropping the runtime (which used to be a source of bugs in the distant past).

### CODE REVIEW

File: tokio/tests/process_issue_42.rs [hunk 1]
```diff
@@ -2,61 +2,39 @@
...
-}
-
-#[test]
-fn issue_42() {
-    let max = 10;
-    for i in 0..max {
-        println!("running {}/{}", i, max);
-        run_test()
-    }
+    timeout(Duration::from_secs(1), join_all(join_handles))
```
Comment (carllerche) [inline tokio/tests/process_issue_42.rs]: Timeouts in general are problematic on CI. It could be that this was the cause for the CI failure.

Comment (ipetkov) [reply tokio/tests/process_issue_42.rs]: I can try taking this out, the test runs pretty fast on my machine at least
